ledger/src/blockstore.rs

```diff
@@ -554,8 +568,10 @@ impl Blockstore {
 
     fn do_open(ledger_path: &Path, options: BlockstoreOptions) -> Result<Blockstore> {
         fs::create_dir_all(&ledger_path)?;
-        let blockstore_path =
-            ledger_path.join(Self::blockstore_directory(&options.shred_storage_type));
+        let blockstore_path = ledger_path.join(Self::blockstore_directory(
+            &options.advanced_options.shred_storage_type,
+        ));
+        let advanced_options = options.advanced_options.clone();
```

Comment:
Is this clone necessary; can we do a move later in creation of `Blockstore` with:
```
advanced_options: options.advanced_options,
```

Reply:
This one is not avoidable because `options` was already moved in the previous `Database::open`.